.github/workflows/check.yaml

```
@@ -18,24 +18,22 @@ jobs:
         python-version: ["3.8", "3.10"]
     with:
       python-version: ${{ matrix.python-version }}
-      tox-version: "<4"
 
   func:
     uses: canonical/bootstack-actions/.github/workflows/func.yaml@v3
     needs: lint-unit
     strategy:
       fail-fast: false
     with:
-      command: "make functional"
-      juju-channel: "2.9/stable"
+      command: "TEST_JUJU3=1 make functional"  # TEST_JUJU3 needed due https://github.com/openstack-charmers/zaza/blob/b22c2eed4c322f1dfc14ffb2d31e0dd18c911a40/setup.py#L47 to support Juju3+
+      juju-channel: "3.4/stable"
       nested-containers: false
       python-version: "3.10"
       timeout-minutes: 120
-      tox-version: "<4"
       action-operator: false
       external-controller: true
       runs-on: "['self-hosted', 'runner-cloudsupport']"
-      juju-controller: soleng-ci-ctrl-29
+      juju-controller: soleng-ci-ctrl-34
       zaza-yaml: "LS0tCm1vZGVsX3NldHRpbmdzOgogIGltYWdlLXN0cmVhbTogcmVsZWFzZWQKcmVnaW9uOiBwcm9kc3RhY2s2CmNsb3VkOiBidWlsZGVyLWNsb3VkLTI5CmNyZWRlbnRpYWw6IGJ1aWxkZXItY2xvdWQtMjktY3JlZAo="
```

```suggestion
      zaza-yaml: "LS0tCm1vZGVsX3NldHRpbmdzOgogIGltYWdlLXN0cmVhbTogcmVsZWFzZWQKcmVnaW9uOiBwcm9kc3RhY2s2CmNsb3VkOiBidWlsZGVyLWNsb3VkCmNyZWRlbnRpYWw6IGJ1aWxkZXItY2xvdWQtY3JlZAo="
```
From the failed func test, I found that we were using the wrong zaza cloud and credential config (specific for 2.9). I've changed the base64 encoded string to reflect the names for the juju 3.x setup.